modules/tide_content_collection/src/Plugin/Field/FieldWidget/ContentCollectionConfigurationWidget.php

```diff
@@ -0,0 +1,1201 @@
+        $default_values = $settings['content']['internal']['field_tags']['default_values'] ?? [];
+        $field_filter = $this->indexHelper->buildEntityReferenceFieldFilter($this->index, 'field_tags', $default_values);
+        if ($field_filter) {
+          $element['settings']['content']['internal']['field_tags']['default_values'] = $field_filter;
+          $element['settings']['content']['internal']['field_tags']['default_values']['#title'] = 'Default tags';
+          $element['settings']['content']['internal']['field_tags']['default_values']['#states'] = [
+            'visible' => [
+              ':input[name="fields[' . $field_name . '][settings_edit_form][settings][settings][content][internal][field_tags][enabled]"]' => ['checked' => FALSE],
+            ],
+          ];
+        }
+      }
+    }
+
+    $entity_reference_fields = $this->getEntityReferenceFields();
+    if (!empty($entity_reference_fields)) {
+      foreach ($entity_reference_fields as $field_id => $field_label) {
+        $element['settings']['content']['internal'][$field_id] = [
+          '#type' => 'details',
+          '#title' => $field_label,
+          '#open' => FALSE,
+          '#collapsible' => TRUE,
+          '#weight' => 2,
+        ];
+        $element['settings']['content']['internal'][$field_id]['enabled'] = [
+          '#type' => 'checkbox',
+          '#title' => $this->t('Enable @field_id', ['@field_id' => $field_id]),
+          '#default_value' => $settings['content']['internal'][$field_id]['enabled'] ?? FALSE,
+        ];
+        $element['settings']['content']['internal'][$field_id]['show_filter_operator'] = [
+          '#type' => 'checkbox',
+          '#title' => $this->t('Show filter operator'),
+          '#default_value' => $settings['content']['internal'][$field_id]['show_filter_operator'] ?? FALSE,
+        ];
+        $default_values = $default_values = $settings['content']['internal'][$field_id]['default_values'] ?? [];
+        $field_filter = $this->indexHelper->buildEntityReferenceFieldFilter($this->index, $field_id, $default_values);
+        if ($field_filter) {
+          $element['settings']['content']['internal'][$field_id]['default_values'] = $field_filter;
+          $element['settings']['content']['internal'][$field_id]['default_values']['#title'] = $this->t('Default @field_id', ['@field_id' => $field_id]);
+          $element['settings']['content']['internal'][$field_id]['default_values']['#states'] = [
+            'visible' => [
+              ':input[name="fields[' . $field_name . '][settings_edit_form][settings][settings][content][internal][' . $field_id . '][enabled]"]' => ['checked' => FALSE],
+            ],
+          ];
+        }
+      }
+    }
+
+    $element['settings']['#element_validate'][] = [$this, 'validateSettings'];
+
+    return $element;
+  }
+
+  /**
+   * Handler #element_validate for the "tabs" form elements in settingsForm().
+   *
+   * Used to set the settings value in a clean structure.
```

Comment:
@joshua-salsadigital where is the `@param` PHPDoc?

Reply:
Added Doc comments